File: java/268. Missing Number/MissingNumber.java
import java.util.ArrayList;

public class MissingNumber {
    public static int missingNumber(int[] nums) {
        int n = nums.length;
        int sum = n*(n+1)/2;

        for(var num : nums) {
            sum -= num;
        }

        return sum;
    }
    public static void main(String[] args) {
        int[] nums = {0,1,2,3,5,6};

        System.out.println(missingNumber(nums));
    }
}

File: c++/268. Missing Number/missing_number.cpp
#include<bits/stdc++.h>
using namespace std;

typedef long long ll;

ll solve(vector<ll>& nums) {
    ll n = nums.size();
    ll sum = n*(n+1)/2;

    for(ll num : nums) {
        sum -= num;
    }

    return sum;
}

int main() {
    std::vector<ll> nums = {0,1,2,4,5,6};

    std::cout<<solve(nums);
}